File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/BrandService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<Map> findBrandList();

    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询
     * @param pageNum 当前第几页
     * @param pageSize 每页显示多少条
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);


    /**
     * 带有条件的分页查询
     * @param pageNum
     * @param pageSize
     * @param searchEntity
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize,TbBrand searchEntity);

    /**
     * 添加品牌
     * @param brand
     * @return
     */
    public Result add(TbBrand brand);

    /**
     * 根据主键修改品牌
     * @param brand
     * @return
     */
    public Result update(TbBrand brand);

    /**
     * 根据主键查询
     * @param id
     * @return
     */
    public TbBrand findById(Long id);

    /**
     * 批量删除
     * @param ids
     */
    public void delete(Long[] ids);
}

File: youlexuan-manager-web/src/main/webapp/js/service/brandService.js
/*自定义service服务层*/
/*在自定义服务中，注入内置服务$http*/
/*service中，定义一些方法，每个方法，都可以向后台发请求*/
/*servcie方法中，不对返回值进行处理，仅仅是发送一个请求*/
app.service('brandService',function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    };
    this.findPage = function (pageNum,pageSize) {
        return $http.get("../brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    };
    this.add = function (entity) {
        return $http.post("../brand/add.do",entity)
    };
    this.update = function (entity) {
        return $http.post("../brand/update.do",entity)
    };
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    };
    this.delete = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids)
    };
    this.search = function (pageNum,pageSize,searchEntity) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,searchEntity);
    }
});